app/controllers/courses_controller.rb

```diff
@@ -69,6 +77,11 @@ def destroy
     end
   end
 
+  def lecture_for_current_course(lecture, course)
```

Comment:
I think this method can just be: `lecture.course == course` although it doesn't look like this method is being used